pkg/k8s/factory_functions.go

```diff
@@ -169,9 +169,25 @@ func EqualV1beta1Ingress(ing1, ing2 *v1beta1.Ingress) bool {
 }
 
 func EqualV2CNP(cnp1, cnp2 *cilium_v2.CiliumNetworkPolicy) bool {
-	return cnp1.Name == cnp2.Name &&
-		cnp1.Namespace == cnp2.Namespace &&
-		comparator.MapStringEquals(cnp1.GetAnnotations(), cnp2.GetAnnotations()) &&
+	if !(cnp1.Name == cnp2.Name && cnp1.Namespace == cnp2.Namespace) {
+		return false
+	}
+
+	// Ignore v1.LastAppliedConfigAnnotation annotation
+	lastAppliedCfgAnnotation1, ok1 := cnp1.GetAnnotations()[v1.LastAppliedConfigAnnotation]
+	lastAppliedCfgAnnotation2, ok2 := cnp2.GetAnnotations()[v1.LastAppliedConfigAnnotation]
+	defer func() {
+		if ok1 && cnp1.GetAnnotations() != nil {
+			cnp1.GetAnnotations()[v1.LastAppliedConfigAnnotation] = lastAppliedCfgAnnotation1
+		}
+		if ok2 && cnp2.GetAnnotations() != nil {
+			cnp2.GetAnnotations()[v1.LastAppliedConfigAnnotation] = lastAppliedCfgAnnotation2
+		}
+	}()
+	delete(cnp1.GetAnnotations(), v1.LastAppliedConfigAnnotation)
```

Comment:
Aren't you modifying a live pointer that is available via the CNP store?

Reply:
It's up to the caller to make sure that won't happen. (It's not happening since we always deepcopy before handling the Add/Del/Update)